### PR TITLE
Checks for valid user before accepting ssh auth key

### LINKED ISSUE #239: Disable user shell account - authorized keys
Currently, disabling a user account in API disables username/password login to shell accounts via the pam_auth module, but SSH authorized key bypasses the PAM process entirely. 

Possible solution: use sshd_config `AuthorizedKeyCommand ` option to validate user account via API first, before checking `.ssh/authorized_keys` as usual

### PR DESCRIPTION
**Pull request by hiliang-cmu**
Resolves #239

### CODE REVIEW

**Comment by hiliang-cmu** [comment]
To test:

Create `user1`, `user2`, `user3`

user1: Attempt normal password login via SSH, correct and incorrect password
user2: Login via SSH, add public key to `~/.ssh/authorized_keys`, attempt to login once again, this time with ssh key
user2: Login via SSH, add public key to `~/.ssh/authorized_keys`. Disable/delete account via admin panel. Attempt to login once again, this time with ssh key

File: ansible/roles/pico-shell/files/sshkey_auth.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+def server_valid_user(user):
+    if not competition_active():
+        return False
+    result = try_login(user)
+    return result == "Incorrect password"
+
+
+def main(user):
+    if not server_valid_user(user):
```
Comment (maverickwoo) [inline ansible/roles/pico-shell/files/sshkey_auth.py]: Just checking: if we deploy this, it seems the user `ubuntu` will not be able to use `authorized_keys` unless it is already in the (web) server?

Comment (hiliang-cmu) [reply ansible/roles/pico-shell/files/sshkey_auth.py]: That's a great point

Comment (hiliang-cmu) [reply ansible/roles/pico-shell/files/sshkey_auth.py]: Resolved in 6fb93cc

**Pull request merged**